hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/SerialConsumer.java

```diff
@@ -141,6 +142,9 @@ public Subscription getSubscription() {
     public void updateSubscription(Subscription newSubscription) {
         rateLimiter.updateSubscription(newSubscription);
         sender.updateSubscription(newSubscription);
+        if (messageReceiver != null) {
```

Comment:
Perhaps for cleaner code, we could check whether this consumer is consuming from the outside when updating subscription and avoid null checks. Also, the private method `startConsuming` can use the field instead of a parameter.